runs/comparisons/xcsf_final.py

```diff
@@ -0,0 +1,275 @@
+        "M_PROBATION": xcs.M_PROBATION,
+        "STATEFUL": xcs.STATEFUL,
+        "SET_SUBSUMPTION": xcs.SET_SUBSUMPTION,
+        "THETA_SUB": xcs.THETA_SUB,
+        "COMPACTION": xcs.COMPACTION,
+        "TELETRANSPORTATION": xcs.TELETRANSPORTATION,
+        "GAMMA": xcs.GAMMA,
+        "P_EXPLORE": xcs.P_EXPLORE,
+        "EA_SELECT_TYPE": xcs.EA_SELECT_TYPE,
+        "EA_SELECT_SIZE": xcs.EA_SELECT_SIZE,
+        "THETA_EA": xcs.THETA_EA,
+        "LAMBDA": xcs.LAMBDA,
+        "P_CROSSOVER": xcs.P_CROSSOVER,
+        "ERR_REDUC": xcs.ERR_REDUC,
+        "FIT_REDUC": xcs.FIT_REDUC,
+        "EA_SUBSUMPTION": xcs.EA_SUBSUMPTION,
+        "EA_PRED_RESET": xcs.EA_PRED_RESET,
+    }
+
+
+def default_xcs_params():
+    xcs = xcsf.XCS(1, 1, 1)
+    return get_xcs_params(xcs)
+
+
+class XCSF(BaseEstimator, RegressorMixin):
+    """
+    Almost a correct sklearn wrapper for ``xcsf.XCS``. For example, it can't yet
+    be pickled and some parameters are missing
+    """
+
+    def __init__(self, random_state, MAX_TRIALS=1000, POP_SIZE=200, NU=5,
+                 P_CROSSOVER=0.8, P_EXPLORE=0.9, THETA_EA=50,
+                 EA_SUBSUMPTION=False, EA_SELECT_TYPE="tournament"):
+        self.random_state = random_state
+
+        self.MAX_TRIALS = MAX_TRIALS
+        self.POP_SIZE = POP_SIZE
+        self.NU = NU
+        self.P_CROSSOVER = P_CROSSOVER
+        self.P_EXPLORE = P_EXPLORE
+        self.THETA_EA = THETA_EA
+        self.EA_SUBSUMPTION = EA_SUBSUMPTION
+        self.EA_SELECT_TYPE = EA_SELECT_TYPE
+
+    def fit(self, X, y):
+        X, y = check_X_y(X, y)
+        # This is required so that XCS does not (silently!?) segfault (see
+        # https://github.com/rpreen/xcsf/issues/17 ).
+        y = y.reshape((len(X), -1))
+
+        random_state = check_random_state(self.random_state)
+
+        xcs = xcsf.XCS(X.shape[1], 1, 1)  # only 1 (dummy) action
+        xcs.seed(random_state.randint(np.iinfo(np.int32).max))
+
+        params = default_xcs_params()
+        configurables = {"MAX_TRIALS": self.MAX_TRIALS,
+                         "POP_SIZE": self.POP_SIZE,
+                         "NU": self.NU,
+                         "P_CROSSOVER": self.P_CROSSOVER,
+                         "P_EXPLORE": self.P_EXPLORE,
+                         "THETA_EA": self.THETA_EA,
+                         "EA_SUBSUMPTION": self.EA_SUBSUMPTION,
+                         "EA_SELECT_TYPE": self.EA_SELECT_TYPE}
+        params.update(configurables)
+        set_xcs_params(xcs, params)
+        print(f"fitting with {configurables}")
+
+        xcs.action("integer")  # (dummy) integer actions
+
+        args = {
+            "min": -1,  # minimum value of a lower bound
+            "max": 1,  # maximum value of an upper bound
+            "spread_min": 0.1,  # minimum initial spread
+            "eta":
+                0,  # disable gradient descent of centers towards matched input mean
+        }
+        xcs.condition("hyperrectangle_csr", args)
```

Comment:
change

Reply:
Moved to its own branch https://github.com/heidmic/suprb-experimentation/tree/xcsf_experiment